### PR TITLE
Alanren/toolsservice hadr

### PR DESCRIPTION
**Pull request by alanrenmsft**
Adding support for HA dashboard view

### CODE REVIEW

File: src/Microsoft.SqlTools.ServiceLayer/AvailabilityGroup/Contracts/AvailabilityReplicaInfo.cs [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+using Microsoft.SqlTools.ServiceLayer.AvailabilityGroup;
+
+namespace Microsoft.SqlTools.ServiceLayer.AvailabilityGroup.Contracts
+{
+    /// <summary>
+    /// a class for storing various properties of availability replicas
+    /// </summary>
+    public class AvailabilityReplicaInfo
+    {
+        public string Name { get; set; }
```
Comment (llali) [inline src/Microsoft.SqlTools.ServiceLayer/AvailabilityGroup/Contracts/AvailabilityReplicaInfo.cs]: comments are missing here also

File: src/Microsoft.SqlTools.ServiceLayer/AvailabilityGroup/Contracts/AvailabilityGroupsRequest.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+        public string OwnerUri { get; set; }
+    }
+
+    /// <summary>
+    /// Availability groups result
+    /// </summary>
+    public class AvailabilityGroupsResult
+    {
+
+        public bool Succeeded { get; set; }
```
[the remark this answers is not in the document]
Comment (alanrenmsft) [reply src/Microsoft.SqlTools.ServiceLayer/AvailabilityGroup/Contracts/AvailabilityGroupsRequest.cs]: done

File: src/Microsoft.SqlTools.ServiceLayer/AvailabilityGroup/Contracts/AvailabilityGroupsRequest.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+
+        public string ErrorMessage { get; set; }
```
[the remark this answers is not in the document]
Comment (alanrenmsft) [reply src/Microsoft.SqlTools.ServiceLayer/AvailabilityGroup/Contracts/AvailabilityGroupsRequest.cs]: done

File: src/Microsoft.SqlTools.ServiceLayer/AvailabilityGroup/Contracts/AvailabilityReplicaInfo.cs (hunk 1 above)
Comment (alanrenmsft) [reply src/Microsoft.SqlTools.ServiceLayer/AvailabilityGroup/Contracts/AvailabilityReplicaInfo.cs]: done

File: src/Microsoft.SqlTools.ServiceLayer/AvailabilityGroup/AvailabilityGroupService.cs [hunk 4]
```diff
@@ -0,0 +1,197 @@
...
+                    using (var sqlConnection = ConnectionService.OpenSqlConnection(connInfo))
+                    {
+                        var serverConnection = new ServerConnection(sqlConnection);
+
+                        Server server = new Server(serverConnection);
+                        result.Succeeded = true;
+                        result.AvailabilityGroups = server.AvailabilityGroups.OfType<SMO.AvailabilityGroup>().Select(ag => CreateAvailabilityGroupInfo(ag)).ToArray();
+                        sqlConnection.Close();
+                    }
+                }
```
[the remark this answers is not in the document]
Comment (alanrenmsft) [reply src/Microsoft.SqlTools.ServiceLayer/AvailabilityGroup/AvailabilityGroupService.cs]: done

File: src/Microsoft.SqlTools.ServiceLayer/AvailabilityGroup/AvailabilityGroupService.cs [hunk 5]
```diff
@@ -0,0 +1,197 @@
...
+                await requestContext.SendResult(result);
+            }
+            catch (Exception e)
+            {
+                await requestContext.SendError(e);
+            }
+        }
+
+        private AvailabilityGroupInfo CreateAvailabilityGroupInfo(SMO.AvailabilityGroup availabilityGroup)
```
[the remark this answers is not in the document]
Comment (alanrenmsft) [reply src/Microsoft.SqlTools.ServiceLayer/AvailabilityGroup/AvailabilityGroupService.cs]: I am adding integration test but need to be run in our VSO with special setup, I will work on the test infrastructure.

**Pull request closed without merging**